fgpyo/sam/__init__.py

```diff
@@ -606,6 +607,110 @@ def __getitem__(
         return self.elements[index]
 
 
+@enum.unique
+class PairOrientation(enum.Enum):
+    """Enumerations of read pair orientations."""
+
+    FR = "FR"
+    """A pair orientation for forward-reverse reads ("innie")."""
+
+    RF = "RF"
+    """A pair orientation for reverse-forward reads ("outie")."""
+
+    TANDEM = "TANDEM"
+    """A pair orientation for tandem (forward-forward or reverse-reverse) reads."""
+
+    @classmethod
+    def build(
+        cls, r1: AlignedSegment, r2: Optional[AlignedSegment] = None
+    ) -> Optional["PairOrientation"]:
+        """Returns the pair orientation if both reads are mapped to the same reference sequence.
+
+        Args:
+            r1: The first read in the template.
+            r2: The second read in the template. If undefined, mate data set upon R1 will be used.
+
+        See:
+            [`htsjdk.samtools.SamPairUtil.getPairOrientation()`](https://github.com/samtools/htsjdk/blob/c31bc92c24bc4e9552b2a913e52286edf8f8ab96/src/main/java/htsjdk/samtools/SamPairUtil.java#L71-L102)
+        """
+
+        if r2 is None:
+            r2_is_unmapped = r1.mate_is_unmapped
+            r2_reference_id = r1.next_reference_id
+        else:
+            r2_is_unmapped = r2.is_unmapped
+            r2_reference_id = r2.reference_id
+
+        if r1.is_unmapped or r2_is_unmapped or r1.reference_id != r2_reference_id:
+            return None
+
+        if r2 is None:
+            if not r1.has_tag("MC"):
+                raise ValueError('Cannot determine proper pair status without R2\'s cigar ("MC")!')
```

Comment:
I think we could probably narrow the cases where we raise this error a bit more.  I.e. if the "r2" start is greater than "r1" end, then we don't technically need the mate cigar.

Reply:
Likewise if r2 is positive strand, we won't need the cigar.

Reply:
Good observation!